multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ValidateDeployParametersStep.java

```diff
@@ -32,6 +40,7 @@ public class ValidateDeployParametersStep extends SyncFlowableStep {
 
     private final ResilientOperationExecutor resilientOperationExecutor = new ResilientOperationExecutor();
     private final ExecutorService fileStorageThreadPool;
+    private final List<FileEntry> extensionDescriptors = new ArrayList<>();
```

Comment:
remove this variable, it is better to return them from the validateExtensionDescriptorFileIds method. If someone reverses the methods execution this will stop working without even noticing

Reply:
It's not good idea to use the step to keep some data specific only for the operation, either use this variable as local one or use external class without using bean mechanism for initialization.